Run CI on ruby 2.4

hfm opened the pull request.

hfm committed:
Support Ruby 2.4

hfm committed:
activesupport 5.x requires ruby 2.2.2+

hfm committed:
fix for Ruby 2.4

in ruby 2.4, puppet_noop doesn't work. when i remove version lock of sshkit & parallel gems, it works.

hanazuki commented on puppet-theatre.gemspec:
```diff
@@ -19,9 +19,6 @@ Gem::Specification.new do |spec|
   spec.require_paths = ['lib']
 
-  spec.add_dependency 'sshkit', '1.9.0'
-  spec.add_dependency 'parallel', '1.10.0'
```
These runtime dependencies should not be removed from gemspec.
Instead we should consider solving the problem by relaxing these version constraints. (I don't remember why I pinned the versions 😅 )

hfm committed:
Revert "fix for Ruby 2.4"

This reverts commit e1a013e192ab722f642651ecf873933c385d8f6b.

hfm committed:
Ruby 2.4 requires newer sshkit and parallel.

in ruby 2.4, puppet_noop doesn't work. when i remove version lock of sshkit & parallel gems, it works.

hanazuki commented:
Thank you
